src/riak_ql_parser.yrl

```diff
@@ -683,6 +685,18 @@ make_insert({identifier, Table}, Fields, Values) ->
      {values, Values}
     ].
 
+make_group_by_hash_fn({identifier,<<"time">>}, [{identifier,_} = Col, {integer,_} = Quantum]) ->
+    {time_fn, Col, Quantum};
+make_group_by_hash_fn({identifier,<<"time">>}, [{integer,_}, {identifier,_}]) ->
+    return_error_flat(
+        "Arguments for GROUP BY time(..) were the wrong way round. Function signature is\n"
+        "  GROUP BY time(<column>, <milliseconds>)\n"
```

Comment:
The format is actually:
`GROUP BY time(<column>, <duration_units><duration_unit>)`